rusty_pizza_server/src/money.rs

```diff
@@ -0,0 +1,383 @@
+use std::fmt::{self, Display, Formatter};
+use std::ops::{Add, Mul, Sub};
+
+#[derive(Debug, PartialEq, Eq, Hash, Copy, Clone)]
+pub struct Money {
+    cents: u32,
+}
+
+impl Money {
+    pub fn new(euros: u32, cents: u8) -> Money {
+        Money {
+            cents: euros * 100 + cents as u32,
+        }
+    }
+
+    pub fn get_euros(&self) -> u32 {
+        self.cents / 100
+    }
+
+    pub fn get_cents(&self) -> u8 {
+        (self.cents % 100) as u8
+    }
+
+    pub fn get_total_cents(&self) -> u32 {
+        self.cents
+    }
+}
+
+impl Add for Money {
+    type Output = Self;
+
+    fn add(self, other: Self) -> Self {
+        Self {
+            cents: self.cents + other.cents,
+        }
+    }
+}
+
+impl Sub for Money {
+    type Output = Self;
+
+    fn sub(self, other: Self) -> Self {
+        Self {
+            cents: self.cents - other.cents,
+        }
+    }
+}
+
+impl Mul<u8> for Money {
+    type Output = Self;
+
+    fn mul(self, other: u8) -> Self {
+        Self {
+            cents: self.cents * other as u32,
+        }
+    }
+}
+
+impl Mul<Money> for u8 {
+    type Output = Money;
+
+    fn mul(self, other: Money) -> Money {
+        other * self
+    }
+}
+
+impl Mul<u16> for Money {
+    type Output = Self;
+
+    fn mul(self, other: u16) -> Self {
+        Self {
+            cents: self.cents * other as u32,
+        }
+    }
+}
+
+impl Mul<Money> for u16 {
+    type Output = Money;
+
+    fn mul(self, other: Money) -> Money {
+        other * self
+    }
+}
+
+impl Mul<u32> for Money {
+    type Output = Self;
+
+    fn mul(self, other: u32) -> Self {
+        Self {
+            cents: self.cents * other,
+        }
+    }
+}
+
+impl Mul<Money> for u32 {
+    type Output = Money;
+
+    fn mul(self, other: Money) -> Money {
+        other * self
+    }
+}
+
+impl Display for Money {
+    fn fmt(&self, f: &mut Formatter) -> fmt::Result {
+        let euros = self.cents / 100;
+        let cents = self.cents % 100;
+        write!(f, "{},{}€", euros, cents)
```

Comment:
Use the getter methods implemented above.

Reply:
Done.